pkg/ovs/ovsctl/ovsctl_test.go

```diff
@@ -0,0 +1,520 @@
+			got, err := c.GetDPFeatures()
+			require.NoError(t, err)
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func TestTrace(t *testing.T) {
+	srcMAC, _ := net.ParseMAC("11:11:11:11:11:11")
+	dstMAC, _ := net.ParseMAC("11:11:11:11:11:22")
+	srcIP := net.ParseIP("1.1.1.2")
+	dstIP := net.ParseIP("2.2.2.2")
+	tc := []struct {
+		name          string
+		req           *TracingRequest
+		expectedFlow  string
+		expectedError string
+	}{
+		{
+			name: "Trace IP Packet-success",
+			req: &TracingRequest{
+				InPort:              "3",
+				SrcIP:               srcIP,
+				DstIP:               dstIP,
+				SrcMAC:              srcMAC,
+				DstMAC:              dstMAC,
+				Flow:                "in_port=3,tcp,tcp_dst=22,",
+				AllowOverrideInPort: true,
+			},
+			expectedFlow: "dl_src=11:11:11:11:11:11,dl_dst=11:11:11:11:11:22,in_port=3,tcp,tcp_dst=22,,nw_ttl=64,nw_src=1.1.1.2,nw_dst=2.2.2.2,",
+		},
+		{
+			name: "no-ops due to source and destination IP error for non-IP packet",
+			req: &TracingRequest{
+				InPort: "9",
+				SrcIP:  srcIP,
+				Flow:   "table=10, n_packets=0, n_bytes=0, priority=200,arp,actions=resubmit(,20)",
+			},
+			expectedError: "source and destination must not be specified for non-IP packet",
+		},
+		{
+			name: "no-ops due to duplicated source IP error",
+			req: &TracingRequest{
+				InPort: "9",
+				SrcIP:  srcIP,
+				Flow:   "nw_src=1.1.1.2,tcp,tcp_dst=22,",
+			},
+			expectedError: "duplicated 'nw_src' in flow",
+		},
+		{
+			name: "no-ops due to duplicated destination IP error",
+			req: &TracingRequest{
+				InPort: "9",
+				DstIP:  dstIP,
+				Flow:   "nw_dst=2.2.2.2,tcp,tcp_dst=22,",
+			},
+			expectedError: "duplicated 'nw_dst' in flow",
+		},
+		{
+			name: "no-ops due to duplicated in_port error",
+			req: &TracingRequest{
+				InPort:              "9",
+				Flow:                "in_port=3,tcp,tcp_dst=22,",
+				AllowOverrideInPort: false,
+			},
+			expectedError: "duplicated 'in_port' in flow",
+		},
+	}
+	for _, tt := range tc {
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			mockOVSAppctlRunner := NewMockOVSAppctlRunner(ctrl)
+			client := &ovsCtlClient{
+				bridge:          "br-int",
+				ovsAppctlRunner: mockOVSAppctlRunner,
+			}
+			if tt.expectedFlow != "" {
+				mockOVSAppctlRunner.EXPECT().RunAppctlCmd("ofproto/trace", true, tt.expectedFlow)
+			}
+			_, err := client.Trace(tt.req)
+			if tt.expectedError != "" {
+				assert.EqualError(t, err, tt.expectedError)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestDeleteDPInterface(t *testing.T) {
+	tc := []struct {
+		name            string
+		datapath        []byte
+		updatedDatapath []byte
+		expectedError   string
+	}{
+		{
+			name:            "1",
+			datapath:        testDatapath1,
+			updatedDatapath: testDatapath2,
+			expectedError:   "",
+		},
+		{
+			name:          "2",
+			datapath:      testDatapath2,
+			expectedError: "no such device",
+		},
+	}
+
+	for _, tt := range tc {
```

Comment:
I don't think you should use table-driven tests here. Rather define 2 subtests like this:
```go
t.Run("delete interface", fun(t *testing.T) {
    // ...
}
t.Run("unknown interface", fun(t *testing.T) {
    // ...
}
```

The test is not written in a very intuitive way at the moment. "1" and "2" are not very good test case names, and I can't understand the name `updatedDatapath` in the context of this test.